src/core/global_config.rs

```diff
@@ -146,6 +240,145 @@ impl GlobalConfig {
 
         Ok(())
     }
+
+    fn assert_no_errors_in_projects(errors: &Vec<String>) -> Result<(), String> {
+        if !errors.is_empty() {
+            return Err(errors.join(", "));
+        }
+        Ok(())
+    }
+
+    fn assert_no_duplicate_project_name(projects: &Vec<Project>) -> Result<(), String> {
+        let project_count_by_name = projects
+            .iter()
+            .map(|project| project.get_name())
+            .fold(std::collections::HashMap::new(), |mut acc, name| {
+                *acc.entry(name).or_insert(0) += 1;
+                acc
+            })
```

Comment:
I agree it will be nice to refactor the code, but it seems like the project name would be hard to pass in the error message. 

Maybe it will be of no use to have it in the message (as if we had a project, we should have its name and thus know which project throws this error), but if we try in the futur to implement a command like `heal` to verify the config is right it could be useful

Reply:
As you wish. The example I gave compiles and shall work for the three copies of this fucntions (assert_no_duplicate_*), as it just accept the result of a map on projects, or devices, ...